tests/accessor_tests.cpp

```diff
@@ -0,0 +1,177 @@
+	static constexpr auto enum_component_type = ComponentType::Float;
+};
+
+static const std::byte* getBufferData(const fastgltf::Buffer& buffer) {
+	const std::byte* result = nullptr;
+
+	std::visit(fastgltf::visitor {
+		[](auto&) {},
+		[&](const fastgltf::sources::Vector& vec) {
+			result = reinterpret_cast<const std::byte*>(vec.bytes.data());
+		},
+		[&](const fastgltf::sources::ByteView& bv) {
+			result = bv.bytes.data();
+		},
+	}, buffer.data);
+	
+	return result;
+}
+
+TEST_CASE("Test accessor", "[gltf-tools]") {
+    auto lightsLamp = sampleModels / "2.0" / "LightsPunctualLamp" / "glTF";
+    fastgltf::GltfDataBuffer jsonData;
+    REQUIRE(jsonData.loadFromFile(lightsLamp / "LightsPunctualLamp.gltf"));
+
+    fastgltf::Parser parser(fastgltf::Extensions::KHR_lights_punctual);
+    auto model = parser.loadGLTF(&jsonData, lightsLamp, fastgltf::Options::LoadExternalBuffers);
+    REQUIRE(parser.getError() == fastgltf::Error::None);
+    REQUIRE(model->parse(fastgltf::Category::Buffers | fastgltf::Category::BufferViews
+			| fastgltf::Category::Accessors) == fastgltf::Error::None);
+    REQUIRE(model->validate() == fastgltf::Error::None);
+
+    auto asset = model->getParsedAsset();
+    REQUIRE(asset->accessors.size() == 15);
+    auto& accessors = asset->accessors;
+
+    SECTION("getAccessorElement<std::uint16_t>") {
+        auto& firstAccessor = accessors[0];
+		REQUIRE(firstAccessor.type == fastgltf::AccessorType::Scalar);
+		REQUIRE(firstAccessor.componentType == fastgltf::ComponentType::UnsignedShort);
+
+		REQUIRE(firstAccessor.bufferViewIndex.has_value());
+		auto& view = asset->bufferViews[*firstAccessor.bufferViewIndex];
+
+		auto* bufferData = getBufferData(asset->buffers[view.bufferIndex]);
+		REQUIRE(bufferData != nullptr);
+
+		auto* checkData = reinterpret_cast<const std::uint16_t*>(bufferData + view.byteOffset
+				+ firstAccessor.byteOffset);
+
+		REQUIRE(*checkData == fastgltf::getAccessorElement<std::uint16_t>(*asset, firstAccessor, 0));
+    }
+
+	{
+        auto& secondAccessor = accessors[1];
+		REQUIRE(secondAccessor.type == fastgltf::AccessorType::Vec3);
+		REQUIRE(secondAccessor.componentType == fastgltf::ComponentType::Float);
+
+		REQUIRE(secondAccessor.bufferViewIndex.has_value());
+		auto& view = asset->bufferViews[*secondAccessor.bufferViewIndex];
+
+		auto* bufferData = getBufferData(asset->buffers[view.bufferIndex]);
+		REQUIRE(bufferData != nullptr);
+
+		auto* checkData = reinterpret_cast<const glm::vec3*>(bufferData + view.byteOffset
+				+ secondAccessor.byteOffset);
+
+		SECTION("getAccessorElement<glm::vec3>") {
+			REQUIRE(*checkData == fastgltf::getAccessorElement<glm::vec3>(*asset, secondAccessor, 0));
+		}
+
+		SECTION("iterateAccessor") {
+			auto dstCopy = std::make_unique<glm::vec3[]>(secondAccessor.count);
+			std::size_t i = 0;
+
+			fastgltf::iterateAccessor<glm::vec3>(*asset, secondAccessor, [&](auto&& v3) {
+				dstCopy[i++] = std::move(v3);
```

Comment:
Use `std::forward`.

Reply:
Fixed in [df17edc](https://github.com/spnda/fastgltf/pull/20/commits/df17edc742f1893ece80c0cc2467a9b68b7ec880)